### DISCUSSION TITLE
Possible celery beat bug adding task with long future start date and period

### DISCUSSION THREAD

**Discussion by nev7n** [celery/celery#7763] [Q&A] [answered]
Posting here as directed by new issue checklist before raising issue.

We are using celery 5.2.6 as a scheduler in our python flask application, with celerybeat_mongo 0.2.0

We have around 20 tasks that are scheduled to run daily, and one task that we created to start a long time in the future with a long period:
```
"start_after" : ISODate("2032-09-11T05:29:00.365Z")
"interval" : {
                "_cls" : "Interval",
                "every" : 3650,
                "period" : "days"
        },
```

For some reason the task has last_run_at timestamp matching it's creation time, despite run_immediately being false:
```
"last_run_at" : ISODate("2022-09-14T05:29:05.144Z")
"run_immediately" : false
```

After creating this task no other scheduled tasks would run.  After deleting this task the other tasks ran again as scheduled.  

We can see in our logs:
```
[2022-09-16 10:12:41,840: DEBUG/MainProcess] beat: Waking up in 10.00 seconds. 
```
which I think indicates that the celery beat tick() function was being called, but it was choosing not to start any tasks.

**Comment by auvipy** [comment, marked as answer]
can you cross check if its a celery issue or the library you are using, please?

**Reply by nev7n** [reply]
@auvipy I've tried to recreate the problem directly in celery without using the celerybeat_mongo library but have not been able to, so I think this is an issue with the library.
